pkg/storage/tpr/storage_interface_test.go

```diff
@@ -17,12 +17,70 @@ limitations under the License.
 package tpr
 
 import (
+	"context"
+	"reflect"
 	"testing"
 
 	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	_ "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/install"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/testapi"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/diff"
 )
 
+const (
+	namespace = "testns"
+	name      = "testthing"
+)
+
+func TestCreate(t *testing.T) {
+	iface, fakeCl := getTPRStorageIFaceAndFakeCoreRESTClient(t)
+	broker := &v1alpha1.Broker{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1alpha1.SchemeGroupVersion.String(),
+			Kind:       ServiceBrokerKind.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+	}
+	outBroker := &v1alpha1.Broker{}
+	if err := iface.Create(
+		context.Background(),
+		name,
+		broker,
+		outBroker,
+		uint64(0),
+	); err != nil {
+		t.Fatalf("error on create (%s)", err)
+	}
+	// Confirm resource version got set during the create operation
+	if outBroker.ResourceVersion == "" {
+		t.Fatalf("resource version was not set as expected")
+	}
+	// Confirm the output is identical to what is in storage (nothing funny
+	// happened during encoding / decoding the response).
+	obj := fakeCl.storage.get(name, ServiceBrokerKind.URLName(), name)
```

Comment:
check out the in-memory storage behind the fake client. the map replaces the namespace with name